plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1959,6 +1959,78 @@ public void testHistoryTable()
         }
     }
 
+    @Test
+    public void testHistoryTableWithDeletedTransactionLog()
+            throws Exception
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_history_table_with_deleted_transaction_log",
+                "(int_col INTEGER)")) {
+
+            for (int i = 1; i <= 8; i++) {
+                assertUpdate("INSERT INTO " + table.getName() + " VALUES (" + i + ")", 1);
+            }
+            // Add more transaction to generate checkpoint file
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE int_col = 1", 1);
+            assertUpdate("UPDATE " + table.getName() + " SET int_col = int_col * 2 WHERE int_col = 6", 1);
+
+            assertQuery(
+                    "SELECT version, operation FROM \"" + table.getName() + "$history\"",
+                    "VALUES (0, 'CREATE TABLE'), (1, 'WRITE'), (2, 'WRITE'), (3, 'WRITE'), (4, 'WRITE'), (5, 'WRITE'), (6, 'WRITE'), (7, 'WRITE'), (8, 'WRITE'), (9, 'MERGE'), (10, 'MERGE')");
+            assertQuery(
```

Comment:
Why are there so many $history queries ?
Shouldn't we be interested only on what gets retrieved before doing the deletion of the transaction log entries and what is happening after we remove a few of the transaction log files?
Isn't enough simply querying $history table:
- without any filter
- with filter version > 0
- with filter version < checkpointVersion+1
- with filter version BETWEEN 0 and checkpointVersion+1
?